### PR TITLE
CommonClient: fix bug when using Connect button without a disconnect

### PR DESCRIPTION
**Pull request by qwint**
## What is this fixing or adding?
makes the kivy connect button do the same username forgetting that /connect does to fix an issue where losing connection would make you unable to connect to a different server

## How was this tested?
With two servers that have different valid slot names

* connect to server1, enter slot name when prompted
* close server1 to simulate the server losing connection for any reason
* enter server2 server address (with just host:port), click Connect button
* notice without the fix the connection is rejected for invalid slot (even without a prompt)
* notice with the fix the client prompts for slot name and can connect without issue after inputting it

### CODE REVIEW

File: kvui.py [hunk 1]
```diff
@@ -599,6 +599,7 @@ def connect_button_action(self, button):
             self.ctx.username = None
             async_start(self.ctx.disconnect())
         else:
+            self.ctx.username = None
```
Comment (Jouramie) [inline kvui.py:602]: Since `self.ctx.username = None` is in both branch of the `if self.ctx.server`, would it be better to extract it?

**Review by Jouramie — APPROVED** [review]
Maybe it could be considered as another bug (or not a bug at all), but I see that `ctx.connect` also set `ctx.password`. Should it also set to `None`? 

Anyway, given this at least fix the issue when there is no password, LGTM.

**Comment by qwint** [comment]
> Maybe it could be considered as another bug (or not a bug at all), but I see that `ctx.connect` also set `ctx.password`. Should it also set to `None`?
> 
> Anyway, given this at least fix the issue when there is no password, LGTM.

invalid passwords trigger a different workflow where you can fix it, nothing (in that specific workflow) for clearing and fixing a slot, so I think that's fine

File: kvui.py (hunk 1 above)
Comment (qwint) [reply kvui.py]: didn't notice this haha, fixed

**Review by Exempt-Medic — APPROVED** [review]
Merged in and texted the changes which fixed a bug that consistently annoyed me

**Review by black-sliver — APPROVED** [review]
Change looks safe and I am gonna trust the 2 peer reviews.

The password thing is addressed in #3641

**Pull request merged**